cluster-autoscaler/context/autoscaling_context.go

```diff
@@ -42,6 +43,8 @@ type AutoscalingContext struct {
 	PredicateChecker *simulator.PredicateChecker
 	// ExpanderStrategy is the strategy used to choose which node group to expand when scaling up
 	ExpanderStrategy expander.Strategy
+	// EstimatorBuilder is the builder function for node count estimator to be used.
+	EstimatorBuilder estimator.EstimatorBuilder
```

Comment:
The reason this is a builder and not the estimator is because BasicEstimator has internal state. Re-using the same object rather than constructing a new one in each iteration (which is how it currently works) would go beyond refactoring IMO.